Src/Web/Web.Shared.Net/HttpContextBaseExtension.cs

```diff
@@ -0,0 +1,26 @@
+﻿namespace System.Web
+{
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Web.Implementation;
+
+    /// <summary>
+    /// HttpContextBaseExtension class provides extensions methods for accessing Web Application Insights objects.
+    /// </summary>
+    public static class HttpContextBaseExtension
+    {        
+        /// <summary>
+        /// Provide access to request generated by Web Application Insights SDK.
+        /// </summary>
+        /// <param name="context">HttpContextBase instance.</param>
+        /// <returns>Request telemetry instance or null.</returns>
+        public static RequestTelemetry GetRequestTelemetry(this HttpContextBase context)
+        {
+            if (context == null || context.ApplicationInstance == null || context.ApplicationInstance.Context == null)
+            {
+                return null;
+            }
+
+            return context.ApplicationInstance.Context.Items[RequestTrackingConstants.RequestTelemetryItemName] as RequestTelemetry;
```

Comment:
Is there a case when ApplicationInstance will throw NotImplmented if not overridden as per [MSDN](https://msdn.microsoft.com/en-us/library/system.web.httpcontextbase.applicationinstance(v=vs.110).aspx)? Or it's always overridden and implemented in the scenarios AI used in?